keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/MetropolisHastings.java

```diff
@@ -142,6 +142,7 @@ public void sample(Map<Long, List<?>> samplesByVertex) {
 
         @Override
         public NetworkState sample() {
+            step();
```

Comment:
The doc in `SamplingAlgorithm#sample` is unclear to me.
```
    /**
     * @return a network state that represents the value of vertices at the
     * end of the algorithm step
     */
```
Whether or not it's being used to sample at the end of the algorithm step is outside of its scope, even more so if we add this in. Maybe it'll be better to change `NetworkSamplesGenerator#stream` to take one more step?